icu4c/source/i18n/usearch.cpp

```diff
@@ -562,6 +562,26 @@ inline void initialize(UStringSearch *strsrch, UErrorCode *status)
     strsrch->pattern.defaultShiftSize = 0;
 }
 
+// If the caller provided a character breakiterator we'll return that,
+// otherwise we lazily create the internal break iterator. 
+static UBreakIterator* getBreakIterator(UStringSearch *strsrch, UErrorCode *status)
+{
+    if (strsrch->search->breakIter != NULL) {
```

Comment:
optional: Would you mind using nullptr in new code?

Reply:
I think I'll actually do a mass NULL -> nullptr change for this whole file, as it seems odd to leave it in a mixed state that is ~50/50% NULL and nullptr.